modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiClient.kt.mustache

```diff
@@ -326,7 +333,16 @@ import com.squareup.moshi.adapter
             headers.forEach { header -> addHeader(header.key, header.value) }
         }.build()
 
-        val response = client.newCall(request).execute()
+        val response: Response = suspendCoroutine { continuation ->
```

Comment:
@sheepdreamofandroids I think here we should use a `suspendCancellableCoroutine` instead of `suspendCoroutine`.

https://github.com/coil-kt/coil/blob/0af5fe016971ba54518a24c709feea3a1fc075eb/coil-base/src/main/java/coil/util/Extensions.kt#L45-L51

Reply:
I did this except for the check on call.isCancelled.

I added checks around the coroutine imports. The other imports may not be necessary all the time but they come from the same very common dependencies as other unconditional imports. I prefer to keep the template simple.

Reply:
The additional value is that an ongoing http call will be cancelled in case the result is not needed anymore. In itself that is valuable.

It seems to me that the time window where you can cancel an ongoing http call is very short, unless there is a huge request body so it will rarely make a difference.

The additional complexity is quite low though so my gut feeling says it's worth it.

Reply:
Thanks 👍